etl/steps/data/garden/energy/2022-12-13/electricity_mix.py

```diff
@@ -0,0 +1,324 @@
+    )
+
+    # Sanity check.
+    error = "Total electricity share does not add up to 100%."
+    assert all(abs(combined["total_share_of_electricity__pct"].dropna() - 100) < 0.01), error
+
+    # Remove unnecessary columns.
+    combined = combined.drop(columns=["total_share_of_electricity__pct"])
+
+    return combined
+
+
+def prepare_output_table(combined: pd.DataFrame) -> catalog.Table:
+    """Convert the combined (BP + Ember) dataframe into a table with the appropriate metadata and variables metadata.
+
+    Parameters
+    ----------
+    combined : pd.DataFrame
+        BP's Statistical Review combined with Ember's Combined Electricity, after adding per capita variables and
+        share variables.
+
+    Returns
+    -------
+    table : catalog.Table
+        Original data in a table format with metadata.
+
+    """
+    # Set an appropriate index and sort rows and columns conveniently.
+    combined = combined.set_index(["country", "year"], verify_integrity=True).sort_index().sort_index(axis=1)
+
+    # Convert dataframe into a table (with no metadata).
+    table = catalog.Table(combined)
+
+    # Load metadata from yaml file.
+    table.update_metadata_from_yaml(METADATA_PATH, DATASET_SHORT_NAME)
+
+    return table
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load data.
+    #
+    # Load BP's statistical review dataset.
+    ds_bp = catalog.Dataset(BP_DATASET_PATH)
+    # Select main table.
+    table_bp = ds_bp["statistical_review"]
+    # Create a convenient dataframe.
+    df_bp = pd.DataFrame(table_bp)
+
+    # Idem for Ember's combined electricity.
+    ds_ember = catalog.Dataset(EMBER_DATASET_PATH)
+    table_ember = ds_ember["combined_electricity"]
+    df_ember = pd.DataFrame(table_ember)
+
+    #
+    # Process data.
+    #
+    # Prepare BP and Ember data.
+    df_bp = process_bp_data(table_bp=table_bp)
+    df_ember = process_ember_data(table_ember=table_ember)
+
+    # Combine both tables, giving priority to Ember data (on overlapping values).
+    combined = combine_two_overlapping_dataframes(df1=df_ember, df2=df_bp, index_columns=["country", "year"])
+
+    # Add carbon intensities.
+    # There is already a variable for this in the Ember dataset, but now that we have combined
+    # BP and Ember data, intensities should be recalculated for consistency.
+    combined["co2_intensity__gco2_kwh"] = (combined["total_emissions__mtco2"] * MT_TO_G) / (
+        combined["total_generation__twh"] * TWH_TO_KWH
+    )
+
+    # Add per capita variables.
+    combined = add_per_capita_variables(combined=combined)
+
+    # Add "share" variables.
+    combined = add_share_variables(combined=combined)
+
+    # Prepare output table.
+    table = prepare_output_table(combined=combined)
+
+    #
+    # Save outputs.
+    #
+    ds_garden = catalog.Dataset.create_empty(dest_dir)
+    # Import metadata from the metadata yaml file.
+    ds_garden.metadata.update_from_yaml(METADATA_PATH, if_source_exists="replace")
```

Comment:
You could use the new method: `ds_garden.update_metadata(METADATA_PATH)`. This also affects the building of table in `prepare_output_table`.

Note that you should call this method after adding the table to the dataset.

Reply:
True, but I'll avoid these changes for now.